ml/dqn_encoders.py

```diff
@@ -0,0 +1,75 @@
+from abc import abstractmethod
+from typing import Tuple
+
+import torch
+from torch import Tensor
+
+from ml import BaseEmbedding
+from ml.encoders import TorchEncoder, TowerEncoder
+from ml.utils import TensorWithMask
+
+
+class BaseQNetwork(TorchEncoder, config_name='base_q_network'):
+    @abstractmethod
+    def forward(
+            self,
+            current_node_idx: Tensor,
+            neighbor_node_ids: TensorWithMask,
+            destination_node_idx: Tensor
+    ) -> Tuple[Tensor, Tensor]:
+        raise NotImplementedError
+
+
+class TowerQNetwork(BaseQNetwork, config_name='tower_q_network'):
```

Comment:
Generic question: why do you need different tower encoders for PPO and DQN? I think it can be implemented as a common encoder which can be suitable for DQN, PPO, and other different approaches.

Reply:
There may be little difference in implementation, so may be leave two different classes?, may be refactor something during integration with env

Reply:
Ok, this was just a suggestion :)